packages/js/product-editor/src/blocks/product-fields/images/edit.tsx

```diff
@@ -162,10 +162,12 @@ export function ImageBlockEdit( {
 	} ) {
 		recordEvent( 'product_images_replace_image_button_click' );
 
-		if (
-			Array.isArray( propertyValue ) &&
-			! propertyValue.some( ( img ) => media.id === img.id )
-		) {
+		if ( Array.isArray( propertyValue ) ) {
+			// Ignore the media if it is replaced by itseft.
+			if ( propertyValue.some( ( img ) => media.id === img.id ) ) {
+				return;
```

Comment:
Fixes the issue, nice!
Ideally we would show a notice letting the user know that the image was already added.